### ISSUE TITLE
False Negative with Textarea

### ISSUE THREAD

**Issue by KevinHerklotz** [evcohen/eslint-plugin-jsx-a11y#470]
In my React component I have the following textarea with a label surrounding it:

```
<label htmlFor="myTextarea">My Textarea:
    <textarea name="myTextarea" id="myTextarea" />
</label>
```

I am using [airbnb rules](https://github.com/airbnb/javascript/blob/eslint-config-airbnb-v17.0.0/packages/eslint-config-airbnb/rules/react-a11y.js) which are:

```
'jsx-a11y/label-has-for': ['error', {
      components: [],
      required: {
          every: ['nesting', 'id'],
      },
      allowChildren: false,
  }],
```

I'm getting the following error: 

> Form label must have ALL of the following types of associated control: nesting, id

... but I'm expecting no error.

I'm using eslint-plugin-jsx-a11y version 6.1.1 and eslint version 5.3.0

**Comment by ljharb** [comment]
This is definitely a bug.

**Comment by rpellerin** [comment]
Experiencing this as well for a nested <input>
![image](https://user-images.githubusercontent.com/5323445/44466404-837d6e80-a620-11e8-895e-b89780fd4430.png)

**Comment by ljharb** [comment]
@rpellerin if you’re using airbnb’s config, you also need to link them with for/id. If not. Please file a new issue.

**Comment by rijulg** [comment]
@ljharb When can we start using this fix? I believe it's not there in 6.1.1 and there's no later version available on npm

**Comment by ljharb** [comment]
@rijulg try v6.1.2
